src/python/pants/backend/go/goals/test.py

```diff
@@ -121,8 +139,36 @@ def transform_test_args(args: Sequence[str]) -> tuple[str, ...]:
 async def run_go_tests(
     field_set: GoTestFieldSet, test_subsystem: TestSubsystem, go_test_subsystem: GoTestSubsystem
 ) -> TestResult:
-    maybe_pkg_info = await Get(
-        FallibleFirstPartyPkgInfo, FirstPartyPkgInfoRequest(field_set.address)
+    maybe_pkg_info, wrapped_target, test_sources = await MultiGet(
+        Get(FallibleFirstPartyPkgInfo, FirstPartyPkgInfoRequest(field_set.address)),
+        Get(WrappedTarget, Address, field_set.address),
+        Get(SourceFiles, SourceFilesRequest([field_set.sources])),
+    )
+    target = wrapped_target.target
+    test_sources_dirpath = field_set.address.spec_path
+
+    explicit_dependencies = await Get(
```

Comment:
Is there a better way to just get direct non-source dependencies?

Reply:
non-source? This should use `await Get(Addresses, DependenciesRequest)`. 

`ExplicitlyProvidedDependencies` is only to see what was literally defined in the BUILD file for the sake of disambiguating dependency inference. Not relevant here.